.github/workflows/failure_free_docs.yml

```
@@ -0,0 +1,29 @@
+name: failure_free_docs
+on:
+  pull_request:
+    paths:
+      - "docs/fail/**"
+
+permissions:
+  contents: read
+
+jobs:
+  checks_docs_for_failures:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+      - name: Check PR for failures
+        run: |
+          FAILING=$(find ./docs/fail -type f)
+          if [ -z "$FAILING" ]; then
+            echo "No failing docs found"
+          else
+            echo "ERROR:"
+            echo "Failing docs found. Delete the files and retry generation."
+            echo "If the failure persists, notify the language owner"
+            echo
+            echo "$FAILING"
```

The PR will already have the diff of failing files (since the files won't be on `main`, because of the check being in place), so I don't think we need to list them in the error?

```suggestion
            echo "ERROR: Failing docs found in the ./docs/fail/ directory!"
            echo "Try retriggering the rundoc job to see if the failures are intermittent."
            echo "If the failure persists, notify the language owner."
```